common/helpers/checksum.c

```diff
@@ -0,0 +1,37 @@
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "checksum.h"
+#include "common/kerl/hash.h"
+#include "common/sign/v1/iss_kerl.h"
+#include "common/trinary/trit_tryte.h"
+
+
+#define RADIX 3
+
+char* iota_checksum(const char* input, const size_t inputLength,
+                    const size_t checksumLength) {
+  Kerl kerl;
+  init_kerl(&kerl);
+
+  char inputTrits[sizeof(char) * inputLength * RADIX];
+  trit_t trits_hash[HASH_LENGTH];
+  char hash[HASH_LENGTH / RADIX];
+  char* checksumTrytes;
+
+  trytes_to_trits((tryte_t*)input, inputTrits, inputLength);
+  kerl_hash(inputTrits, inputLength * RADIX, trits_hash, &kerl);
+  trits_to_trytes((trit_t*)trits_hash, (tryte_t*)hash, HASH_LENGTH);
```

Comment:
move this after `checksumTrytes` is alloced so you can just write the trytes to there